### ISSUE TITLE
[Bug]: The first requests when streaming (range requests) uses a GET instead of a HEAD, breaks compatibility with dotnet servers

### ISSUE THREAD

**Issue by catmanjan** [mozilla/pdf.js#18512]
### Attach (recommended) or Link to PDF file

https://mozilla.github.io/pdf.js/web/viewer.html?file=https://d2tkmshiozsr4v.cloudfront.net/documents/files/000/041/350/original/ad41798557504b40d26e11b4f059c0945bb5ca14/750614main_NASA_FY_2014_Budget_Estimates-508.pdf?1401953041#disableAutoFetch=true&disableStream=true

### Web browser and its version

Google Chrome latest

### Operating system and its version

Windows 11

### PDF.js version

https://mozilla.github.io/pdf.js/web/viewer.html

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

Go the following link: https://mozilla.github.io/pdf.js/web/viewer.html?file=https://d2tkmshiozsr4v.cloudfront.net/documents/files/000/041/350/original/ad41798557504b40d26e11b4f059c0945bb5ca14/750614main_NASA_FY_2014_Budget_Estimates-508.pdf?1401953041#disableAutoFetch=true&disableStream=true

Open up the network inspector, disable caching and do a refresh so you can see the latest requests

The first request you will see will be a GET with a 200 response, this initiates the streaming process for PDF js

### What is the expected behavior?

Per https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests this first request should be a HEAD request.

### What went wrong?

The latest version of dotnet (8 at the time of writing) has logic in their Kestrel server to prevent Content-Length mismatches when performing GET requests, this is done to avoid a class of exploits known as HTTP request smuggling where an attacked can sneak bytes in at the end of a request.

This means that dotnet servers will not work as efficiently with the streaming feature.

**Comment by catmanjan** [comment]
Is anyone familiar enough with the network layer to tell me if my proposed fix is correct?

I believe this fetch request should be a HEAD not a GET

https://github.com/mozilla/pdf.js/blob/7199c967e68fde48bf2a105296e0b278edc6ffe9/src/display/fetch_stream.js#L130

**Comment by Snuffleupagus** [comment]
Duplicate of #11669

**Comment by NJUST-OpenLib** [comment]
I’ve encountered the same issue.
 I followed an online post and modified the 'disableSteam' and.....disablexxxxxxx setting, 

but the problem persists. In the network panel, I noticed one 200 request (for a complete PDF file) and multiple 206 requests (for partial PDF files). 

This seems inefficient and wasteful in terms of traffic. I

 suspect there might be an error in my settings, but I haven’t found any documentation to address this
